CDP4Common/AutoGenPoco/PossibleFiniteStateList.cs

```diff
@@ -1,279 +1,280 @@
+        /// Note: Ownership in this data model implies the responsibility for the presence and content of this OwnedThing. The owner is always a DomainOfExpertise. The Participant or Participants representing an owner DomainOfExpertise are thus responsible for (i.e. take ownership of) a coherent set of concerns in a concurrent engineering activity.
+        /// </remarks>
+        [UmlInformation(aggregation: AggregationKind.None, isDerived: false, isOrdered: false, isNullable: false, isPersistent: true)]
+        public DomainOfExpertise Owner { get; set; }
+
+        /// <summary>
+        /// Gets or sets a list of ordered contained PossibleFiniteState.
+        /// </summary>
+        /// <remarks>
+        /// ordered collection of finite states for this PossibleFiniteStateList
+        /// </remarks>
+        [UmlInformation(aggregation: AggregationKind.Composite, isDerived: false, isOrdered: true, isNullable: false, isPersistent: true)]
+        public OrderedItemList<PossibleFiniteState> PossibleState { get; protected set; }
+
+        /// <summary>
+        /// Gets an <see cref="IEnumerable{IEnumerable}"/> that references the composite properties of the current <see cref="PossibleFiniteStateList"/>.
+        /// </summary>
+        public override IEnumerable<IEnumerable> ContainerLists
+        {
+            get 
+            {
+                var containers = new List<IEnumerable>(base.ContainerLists);
+                containers.Add(this.PossibleState);
+                return containers;
+            }
+        }
+
+        /// <summary>
+        /// Creates and returns a copy of this <see cref="PossibleFiniteStateList"/> for edit purpose.
+        /// </summary>
+        /// <param name="cloneContainedThings">A value that indicates whether the contained <see cref="Thing"/>s should be cloned or not.</param>
+        /// <returns>
+        /// A cloned instance of <see cref="PossibleFiniteStateList"/>.
+        /// </returns>
+        protected override Thing GenericClone(bool cloneContainedThings)
+        {
+            var clone = (PossibleFiniteStateList)this.MemberwiseClone();
+            clone.Alias = cloneContainedThings ? new ContainerList<Alias>(clone) : new ContainerList<Alias>(this.Alias, clone);
+            clone.Category = new List<Category>(this.Category);
+            clone.Definition = cloneContainedThings ? new ContainerList<Definition>(clone) : new ContainerList<Definition>(this.Definition, clone);
+            clone.ExcludedDomain = new List<DomainOfExpertise>(this.ExcludedDomain);
+            clone.ExcludedPerson = new List<Person>(this.ExcludedPerson);
+            clone.HyperLink = cloneContainedThings ? new ContainerList<HyperLink>(clone) : new ContainerList<HyperLink>(this.HyperLink, clone);
+            clone.PossibleState = cloneContainedThings ? new OrderedItemList<PossibleFiniteState>(clone, true) : new OrderedItemList<PossibleFiniteState>(this.PossibleState, clone);

```

Comment:
No. First case for the constructor 
public OrderedItemList(Thing container, bool isComposite = false)
and the second for
public OrderedItemList(OrderedItemList<T> orderedItemList, Thing container)

Actually to be more efficient this line
clone.PossibleState = cloneContainedThings ? new OrderedItemList<PossibleFiniteState>(clone, true) : new OrderedItemList<PossibleFiniteState>(this.PossibleState, clone);
should be updated to 
clone.PossibleState = cloneContainedThings ? null : new OrderedItemList<PossibleFiniteState>(this.PossibleState, clone);
because for cloneContainedThings=true we reset clone.PossibleState. So it will prevent a creation of a useless object.